functions/src/index.ts

```diff
@@ -166,3 +177,141 @@ export const loadFirestore = functions.https.onRequest(
     response.status(200).send();
   }
 );
+
+/*
+ * Listen for changes to the User document
+ * and initiate the appropriate Glean ping(s)
+ */
+export const handleUserChangesImpl = async function (
+  change: Change<DocumentSnapshot>,
+  context: EventContext
+): Promise<boolean> {
+  const userID = context.params.userID;
+  const rallyID = await getRallyIdForUser(userID);
+  if (!rallyID) {
+    // Without Rally ID, we can't make any Glean pings
+    // This is bad and should be flagged for inspection
+    throw new Error(
+      `Unable to obtain Rally ID for user ID ${userID}. Aborting Glean ping process.`
+    );
+  }
+
+  // Get an object with the current document value.
+  // If the document does not exist, it has been deleted.
+  const newUser = change.after.exists ? change.after.data() : null;
+
+  // Get the old document, to compare the enrollment state.
+  const oldUser = change.before.exists ? change.before.data() : null;
```

Comment:
I think you could just do something similar to the pattern you use for determining the studyID elsewhere, wdyt?

```suggestion
  // Get an object with the current document value.
  // If the document does not exist, it has been deleted.
  const newUser = change.after.exists ? change.after.data() : null;

  // Get the old document, to compare the enrollment state.
  const oldUser = change.before.exists ? change.before.data() : null;

  const rallyID =
  (newUser ? newUser.rallyId : null) ||
  (oldUser ? oldUser.rallyId : null);

```

Reply:
I suppose throwing here if the rally ID does not exist would be good. Also it'd be nice if we got consistent with `rallyID` vs. `rallyId`, we could change that in Firestore if you want.